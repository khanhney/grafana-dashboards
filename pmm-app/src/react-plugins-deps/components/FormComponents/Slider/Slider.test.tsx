import React from 'react';
import { Form } from 'react-final-form';
import { shallow } from 'enzyme';
import { SliderField } from './Slider';

describe('Slider', () => {
  it('Slider renders correct without props', () => {
    const marks = {
      0: 'Low',
      1: 'Medium',
      2: 'High',
    };

    const onSubmit = jest.fn();
    const component = shallow(
      <Form
        onSubmit={onSubmit}
        render={() => <SliderField marks={marks} name="metrics_resolutions_slider" />}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
