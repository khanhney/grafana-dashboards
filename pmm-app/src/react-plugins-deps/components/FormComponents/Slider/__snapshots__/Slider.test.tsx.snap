// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Slider renders correct without props 1`] = `
<ContextProvider
  value={
    Object {
      "batch": [Function],
      "blur": [Function],
      "change": [Function],
      "destroyOnUnregister": false,
      "focus": [Function],
      "getFieldState": [Function],
      "getRegisteredFields": [Function],
      "getState": [Function],
      "initialize": [Function],
      "isValidationPaused": [Function],
      "mutators": Object {},
      "pauseValidation": [Function],
      "registerField": [Function],
      "reset": [Function],
      "resetFieldState": [Function],
      "resumeValidation": [Function],
      "setConfig": [Function],
      "submit": [Function],
      "subscribe": [Function],
    }
  }
>
  <Component
    marks={
      Object {
        "0": "Low",
        "1": "Medium",
        "2": "High",
      }
    }
    name="metrics_resolutions_slider"
  />
</ContextProvider>
`;
